src/graphql/data/nft/Asset.ts

```diff
@@ -0,0 +1,176 @@
+      before: $before
+    ) @connection(key: "AssetQuery_nftAssets") {
+      edges {
+        node {
+          id
+          name
+          ownerAddress
+          image {
+            url
+          }
+          smallImage {
+            url
+          }
+          originalImage {
+            url
+          }
+          tokenId
+          description
+          animationUrl
+          suspiciousFlag
+          collection {
+            name
+            isVerified
+            image {
+              url
+            }
+            creator {
+              address
+              profileImage {
+                url
+              }
+              isVerified
+            }
+            nftContracts {
+              address
+              standard
+            }
+          }
+          listings(first: 1) {
+            edges {
+              node {
+                address
+                createdAt
+                endAt
+                id
+                maker
+                marketplace
+                marketplaceUrl
+                orderHash
+                price {
+                  currency
+                  value
+                }
+                quantity
+                startAt
+                status
+                taker
+                tokenId
+                type
+              }
+              cursor
+            }
+          }
+          rarities {
+            provider
+            rank
+            score
+          }
+          metadataUrl
+        }
+      }
+    }
+  }
+`
+
+const assetQuery = graphql`
+  query AssetQuery(
+    $address: String!
+    $orderBy: NftAssetSortableField
+    $asc: Boolean
+    $filter: NftAssetsFilterInput
+    $first: Int
+    $after: String
+    $last: Int
+    $before: String
+  ) {
+    ...AssetQuery_nftAssets
+  }
+`
+
+export function useAssetsQuery(
+  address: string,
+  orderBy: NftAssetSortableField,
+  asc: boolean,
+  filter: NftAssetsFilterInput,
+  first?: number,
+  after?: string,
+  last?: number,
+  before?: string
+) {
+  const assetsQueryReference = loadQuery<AssetQuery>(RelayEnvironment, assetQuery, {
+    address,
+    orderBy,
+    asc,
+    filter,
+    first,
+    after,
+    last,
+    before,
+  })
+  const queryData = usePreloadedQuery<AssetQuery>(assetQuery, assetsQueryReference)
+  const { data, hasNext, loadNext, isLoadingNext } = usePaginationFragment<AssetPaginationQuery, any>(
+    assetPaginationQuery,
+    queryData
+  )
+
+  const assets: GenieAsset[] = data.nftAssets?.edges?.map((queryAsset: { node: any }) => {
+    const asset = queryAsset.node
+    return {
+      id: asset.id,
+      address: asset.collection.nftContracts[0].address,
+      notForSale: asset.listings === null,
+      collectionName: asset.collection?.name,
+      collectionSymbol: asset.collection?.image?.url,
+      imageUrl: asset.image?.url,
+      animationUrl: asset.animationUrl,
+      marketplace: asset.listings?.edges[0].node.marketplace.toLowerCase(),
```

Comment:
how do you gurantee the existence of edges here (as well nftContracts above and below)?

Reply:
We have null checks around nftContracts which would shown a collection page with no assets and around listings which shows an NFT that is not for sale

Reply:
i more meant that edges was not 0 length, but maybe if listings exists it's guaranteed to be non-zero length?